insertionorderedmap.go

```diff
@@ -0,0 +1,25 @@
+package goflags
+
+type InsertionOrderedMap struct {
+	values map[string]*flagData
+	keys   []string `yaml:"-"`
+}
+
+func (insertionOrderedMap *InsertionOrderedMap) Get(key string) *flagData {
+	value := insertionOrderedMap.values[key]
+	return value
+}
+
+func (insertionOrderedMap *InsertionOrderedMap) forEach(fn func(key string, data *flagData)) {
+	for _, key := range insertionOrderedMap.keys {
+		fn(key, insertionOrderedMap.Get(key))
+	}
+}
+
+func (insertionOrderedMap *InsertionOrderedMap) Set(key string, value *flagData) {
+	_, present := insertionOrderedMap.values[key]
+	insertionOrderedMap.values[key] = value
```

Comment:
shouldn't this be inside the `if !present` block too?

Reply:
It enables you to overwrite the value of an existing mapping and at the same time it prevents duplicating the key in the collection that stores the order

Reply:
Okay, that answers my question.